### DISCUSSION TITLE
关于资源文件的加密

### DISCUSSION THREAD

**Discussion by iostyle** [Tencent/vap#97] [Q&A] [answered]
对于资源文件的加密各位有什么方案吗，比如我不想自己应用内的资源被盗用
不是资源目录，是下载后存储在存储卡内的

**Comment by hexleo** [comment]
可以给几个建议：
1 文件从下载到读取一直处于加密状态，只在内存中进行解码
2 放置到/data/data/your.app.package目录下，至少需要root权限才能获取

**Reply by iostyle** [reply]
嗯是的，第二条考虑到了，但只要对方想获取就意义不大，第一条建议非常不错，感谢分享

**Reply by iostyle** [reply]
@hexleo 在内存中解码为byte[]，startPlay重载只提供了file、asset、IFileContainer，是要自己实现IFileContainer吗请教一下

**Reply by iostyle** [reply]
byte[] 转file后播放不是很安全啊，file会被获取到，可以通过inputStream播放吗

**Reply by iostyle** [reply]
已解决

**Comment by iostyle** [comment, marked as answer]
https://github.com/Tencent/vap/pull/111 已提交PR支持播放ByteArray

**Comment by iostyle** [comment]
总结了一下自己使用的整套方案，如果大家有需要可以参考，欢迎一起探讨
https://github.com/iostyle/EncryptDecryptSolution
